### ISSUE TITLE
Critical error of alt attribute not present is thrown even if image is wrapped inside figure element and a caption is provided in figcaption element.

### ISSUE THREAD

**Issue by manav-sharma69** [dequelabs/axe-core#4473]
### Product

axe Extension

### Product Version

4.80.1

### Latest Version

- [X] I have tested the issue with the latest version of the product

#### Expectation
The extension should not throw warning because `<figcaption>` element was present to provide caption for the image.

#### Actual
The extension threw a `critical` level issue titled *"Images must have alternate text"*.

#### How to Reproduce
The site tested was MDN's demo of [Basic View Transitions](https://mdn.github.io/dom-examples/view-transitions/).

Screenshot:
![alt attribute error](https://github.com/dequelabs/axe-core/assets/157268750/4a07bdba-6aa9-4907-ba91-bf80e1af3762)

#### Additional context
[W3's Guidance for conformance checkers](https://www.w3.org/TR/2014/REC-html5-20141028/embedded-content-0.html#guidance-for-conformance-checkers) says:
- The [img](https://www.w3.org/TR/2014/REC-html5-20141028/embedded-content-0.html#the-img-element) element is in a [figure](https://www.w3.org/TR/2014/REC-html5-20141028/grouping-content.html#the-figure-element) element that satisfies [the conditions described above](https://www.w3.org/TR/2014/REC-html5-20141028/embedded-content-0.html#figcaption-as-alt-condition).
- The condition, in short is: *"In other words, the only content of the [figure](https://www.w3.org/TR/2014/REC-html5-20141028/grouping-content.html#the-figure-element) is an [img](https://www.w3.org/TR/2014/REC-html5-20141028/embedded-content-0.html#the-img-element) element and a [figcaption](https://www.w3.org/TR/2014/REC-html5-20141028/grouping-content.html#the-figcaption-element) element, and the [figcaption](https://www.w3.org/TR/2014/REC-html5-20141028/grouping-content.html#the-figcaption-element) element must include (caption) content."* - [W3](https://www.w3.org/TR/2014/REC-html5-20141028/embedded-content-0.html#figcaption-as-alt-condition)

**Comment by straker** [comment]
Thanks for the issue.

So I tested out the following HTML using different combinations of screen readers and browsers, and this is the result:

```html
<!-- A -->
<figure>
  <img src="https://mdn.github.io/dom-examples/view-transitions/images/jungle-coast.jpg">
  <figcaption>Jungle coast</figcaption>
</figure>

<!-- B -->
<img src="https://mdn.github.io/dom-examples/view-transitions/images/jungle-coast.jpg">

<!-- C -->
<img src="https://mdn.github.io/dom-examples/view-transitions/images/jungle-coast.jpg" alt="cool image of Jungle">
```

* VoiceOver / Safari - When navigating line by line, announces the URL of the image for A and B, and the `alt` for image C. So reading order is: "jungle-coast.jpg image, Jungle coast, jungle-coast.jpg image, cool image of Jungle image." Only when having VoiceOver read the entire page is the figure announced, at which point it's announced as "jungle coast, figure," before it announces the image URL as before.
* JAWS / Chrome - When navigating line by line, skips both images A and B and only announces the `figcaption` as "has details, jungle coast" and image C as "cool image of Jungle graphic." This is the same when the reading the entire page.
* NVDA / Firefox - When navigation line by line, skips both images A and B and only announces the `figcaption` as "figure caption jungle coast" and image C as "graphic cool image of Jungle." This is the same when reading the entire page.

Based on the results it looks like VoiceOver and Safari does not really work with `figure`. Because of this we may call this an accessibility supported issue and it would still be a violation. I'll confirm with the team and let you know if that's not the case.

**Comment by scottaohara** [comment]
that portion of the spec isn't really based on the reality of how images within figures work, both from the time stamp of when that specific version of the spec was referenced (2014) or 10 years later in how reality panned out.

honestly, it's more likely a bug with the html spec in needing to match reality, rather than stating images can have their alts removed, when that doesn't really work well in practice / browsers never implemented anything to account for that text.
